### PR TITLE
Documentation for running Typesense with Docker Swarm

### PR DESCRIPTION
**Pull request by ivishalgandhi**
Adding steps in high-availability.md to deploy Typesense on Docker Swarm

## Change Summary
Update high-availability.md documentation with steps to deploy in Docker Swarm mode

## PR Checklist

- [x] I have read and signed the [Contributor License Agreement](https://forms.gle/PZyiY5N2GDQU8GsV9).

### CODE REVIEW

File: docs-site/content/guide/high-availability.md [hunk 1]
```diff
@@ -272,3 +272,137 @@ final config = Configuration(
 
   </template>
 </Tabs>
+ 
+ ## Deploy a Typesense cluster to Docker Swarm  
+
+To Deploy a Typesense cluster to Docker Swarm, follow these steps:
```
Comment (jasonbosco) [inline docs-site/content/guide/high-availability.md:278]: Could we link to Docker Swarm docs here?

File: docs-site/content/guide/high-availability.md [hunk 2]
```diff
@@ -272,3 +272,137 @@ final config = Configuration(
...
+
+- Initialize Docker Swarm. For example, in a 4 node setup, target docker engine on one of the nodes to act as a manager
```
Comment (jasonbosco) [inline docs-site/content/guide/high-availability.md:280]: Did you mean 3 node setup here?

File: docs-site/content/guide/high-availability.md [hunk 3]
```diff
@@ -272,3 +272,137 @@ final config = Configuration(
...
+Swarm initialized: current node (6082x127zz98f0pwgjexbv5xp) is now a manager.
+   
+To add a worker to this swarm, run the following command:
+
+docker swarm join --token SWMTKN-1-30txqn35hmpwjpk2qq2zmled1rf94pcft2nbhsb0ckleco9pb2-bjh6oh9yz3vk58uimd6v3jjky 192.168.0.25:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+```
+
+- Add docker instance on rest of the nodes to existing Swarm as node. Repeat the below command on all nodes
```
Comment (jasonbosco) [inline docs-site/content/guide/high-availability.md:293]: Could we create named tabs like this so it's clear that these commands need to be run on different nodes: https://github.com/typesense/typesense-website/blob/a11202d03e553826ddf04918cc57396a849f4872/docs-site/content/guide/high-availability.md#example

File: docs-site/content/guide/high-availability.md [hunk 4]
```diff
@@ -272,3 +272,137 @@ final config = Configuration(
...
+- Create a new nodes file on each node that's part of the Swarm 
+
+```shell
+mkdir typesense && cd typesense && echo 'typesense-1:8107:8108,typesense-2:7107:7108,typesense-3:9107:9108' | sudo tee nodes
+```
+
+- Create data folder on all the nodes
+
+ ```shell
+ mkdir /tmp/typesense-data-1/ &&  mkdir /tmp/typesense-data-2/ && mkdir /tmp/typesense-data-3/
```
Comment (jasonbosco) [inline docs-site/content/guide/high-availability.md:310]: Shouldn't this be run on separate nodes, one at a time?

File: docs-site/content/guide/high-availability.md [hunk 5]
```diff
@@ -272,3 +272,137 @@ final config = Configuration(
...
+
+- List the tasks that are part of the deployed stack
+
+```shell 
+docker stack ps ts
+ID             NAME               IMAGE                        NODE      DESIRED STATE   CURRENT STATE           ERROR     PORTS
+x4w38438c7bn   ts_typesense-1.1   typesense/typesense:0.21.0   node2     Running         Running 5 minutes ago
+0iacq1bhw1ia   ts_typesense-2.1   typesense/typesense:0.21.0   node1     Running         Running 5 minutes ago
+wqyec57a3d4o   ts_typesense-3.1   typesense/typesense:0.21.0   node3     Running         Running 5 minutes ago
+```
```
Comment (jasonbosco) [inline docs-site/content/guide/high-availability.md:408]: Could we add a set of curl commands to verify that all nodes are up and clustered successfully?

**Review by jasonbosco — CHANGES_REQUESTED** [review]
After reading through the PR, I think it's best to have this as a separate guide article, instead of putting it under the High Availability guide article. Could you move it to a new article?

File: docs-site/content/guide/high-availability.md (hunk 1 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:278]: Official Docker Swarm documentation?  Sure.

File: docs-site/content/guide/high-availability.md (hunk 2 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:280]: 4 node - 1 node in Docker Swarm as manager and other 3 nodes as workers to host typesense

File: docs-site/content/guide/high-availability.md (hunk 3 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:293]: will do.

File: docs-site/content/guide/high-availability.md (hunk 4 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:310]: can be run in one go

File: docs-site/content/guide/high-availability.md (hunk 5 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:408]: yes will include the Debug API call

File: docs-site/content/guide/high-availability.md (hunk 1 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:278]: will create as a sep markdown doc. i was thinking of same before

File: docs-site/content/guide/high-availability.md (hunk 4 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md:310]: if the service moves to other node the data folder should be present

Comment (jasonbosco) [reply docs-site/content/guide/high-availability.md:310]: The data folder needs to be present, typesense doesn't auto-create it.

**Comment by ivishalgandhi** [comment]
@jasonbosco Please review https://github.com/typesense/typesense-website/pull/111/commits/d0a6778a09df7d838db54e22670e70a4fc2852d5

File: docs-site/content/guide/high-availability.md [hunk 6]
```diff
@@ -272,3 +272,137 @@ final config = Configuration(
 
   </template>
 </Tabs>
+ 
+ ## Deploy a Typesense cluster to Docker Swarm
```
Comment (jasonbosco) [inline docs-site/content/guide/high-availability.md]: Could you remove this section?

**Review by jasonbosco — CHANGES_REQUESTED** [review]
@ivishalgandhi Thank you for putting this together! 

One thing that could potentially confuse users is the distinction between Typesense `nodes` and the Docker `nodes`. I'm wondering if we should be explicit about which node we're referring to when talking about nodes in this document.

File: docs-site/content/guide/high-availability.md (hunk 6 above)
Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md]: done

Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md]: i will change reference to `docker` nodes

Comment (ivishalgandhi) [reply docs-site/content/guide/high-availability.md]: @jasonbosco Please review.

**Review by jasonbosco — APPROVED** [review]

**Pull request merged**